sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp

```diff
@@ -220,6 +220,40 @@ void WinHttpTransport::CreateSessionHandle(std::unique_ptr<_detail::HandleManage
     // ERROR_NOT_ENOUGH_MEMORY
     GetErrorAndThrow("Error while getting a session handle.");
   }
+
+// These options are only available starting Windows 10 Version 2004, starting 06/09/2020
+// These are primarily round trip time (RTT) performance optimizations, and hence if they don't get
+// set successfully, we shouldn't fail the request and continue as if the options don't exist.
+// Therefore, we just log the error to console and move on.
+#ifdef WINHTTP_OPTION_TCP_FAST_OPEN
+  BOOL tcp_fast_open = TRUE;
+  if (!WinHttpSetOption(
+          handleManager->m_sessionHandle,
+          WINHTTP_OPTION_TCP_FAST_OPEN,
+          &tcp_fast_open,
+          sizeof(tcp_fast_open)))
+  {
+    DWORD error = GetLastError();
+    std::wstring message
+        = L"Unable to set TCP Fast Open. Error Code: " + std::to_wstring(error) + L".\n";
+    wprintf(message.c_str());
```

Comment:
?

Reply:
Use logging instead?

Reply:
Why bother logging anything? 
On first try, can't we detect if the system supports this and if not, never try again and just use the normal technique?

Reply:
You don't think it would be useful as a diagnostic for the user to know this detail?

I am fine with not logging it for now.

Reply:
No, I don't think this is worth logging. It's just an if statement and we don't log other if statements. You probably think about logging it because YOU want to know how well this code worked but a customer wouldn't care as there is no action that they can take. If they did, they could put a breakpoint here. 

I would think that there is a way you can ask what version of Windows you're on first and attempt to use this flag if on a version that supports it rather than just trying to use it. Maybe https://docs.microsoft.com/en-us/windows/win32/api/Winbase/nf-winbase-verifyversioninfoa can help?

Reply:
The recommendation from the winhttp folks was to just try and set the option and ignore/move past the error, which is why I didn't check the version up front. It's simpler this way too.

I was thinking, for logging, knowing what type of TCP handshake was being used might be helpful in diagnosis, but I suppose that's more about validating that the underlying transport layer works correctly, which isn't as important compared to what happens at the two ends (client/SDK and server/Azure service).